### PR TITLE
Allow log plot labels to use exponents.

### PR DESCRIPTION
**Pull request by Brow71189**

### CODE REVIEW

**Comment by Brow71189** [comment]
This was a bit more complex than I originally thought. I think it is producing reasonable ticks now.
The log-plot option was pretty broken overall, not only because of the ticker (see also my pull request nion-software/nionswift#391).

**Comment by cmeyer** [comment]
Where does this change show up in practice? Is there data in Swift (like `show(numpy.arange(-2,5,0.01))`) that will highlight the difference before and after this change?

**Comment by cmeyer** [comment]
I simplified the edge case test and modified the handling edge cases when display log plots by always making the ticker produce a single '0'. Here is the diff. Any objection before I merge?

```
diff --git a/nion/utils/Geometry.py b/nion/utils/Geometry.py
index 05988ad..da22d66 100755
--- a/nion/utils/Geometry.py
+++ b/nion/utils/Geometry.py
@@ -8,6 +8,7 @@
 
 # standard libraries
 import collections
+import copy
 import math
 import typing
 
@@ -127,7 +128,17 @@ class Ticker:
         self.__ticks = ticks
         self.__logarithmic = logarithmic
         self.__minimum, self.__maximum, self.__tick_values, self.__division, self.__precision, self.__factor10 = make_pretty_range2(value_low, value_high, ticks=ticks, logarithmic=logarithmic)
-        displayed_tick_values = (math.pow(10.0, tick_value) if logarithmic else tick_value for tick_value in self.__tick_values)
+        # calculate the displayed tick values
+        if logarithmic:
+            # if logarithmic, ensure there are valid values
+            if len(self.__tick_values) > 1:
+                displayed_tick_values = (math.pow(10.0, tick_value) for tick_value in self.__tick_values)
+            # otherwise just use the single value (which will be 0)
+            else:
+                displayed_tick_values = self.__tick_values
+        else:
+            # if not logarithmic, use the tick values
+            displayed_tick_values = tuple(copy.copy(self.__tick_values))
         self.__tick_labels = list(self.value_label(tick_value) for tick_value in displayed_tick_values)
 
     def __nice_label(self, value: float, precision: int, factor10: float) -> str:
diff --git a/nion/utils/test/Geometry_test.py b/nion/utils/test/Geometry_test.py
index 01ca82e..b05766e 100755
--- a/nion/utils/test/Geometry_test.py
+++ b/nion/utils/test/Geometry_test.py
@@ -61,18 +61,14 @@ class TestGeometryClass(unittest.TestCase):
 
     def test_ticker_handles_edge_cases(self):
         for logarithmic in (False, True):
-            def log(val):
-                return math.log10(val) if logarithmic else val
-            with self.subTest(logarithmic=logarithmic):
-                self.assertEqual(Geometry.Ticker(log(1), log(1), logarithmic=logarithmic).labels, ['1'])
-                self.assertEqual(Geometry.Ticker(log(math.nan), log(1), logarithmic=logarithmic).labels, ['1'] if logarithmic else ['0'])
-                if not logarithmic:
-                    self.assertEqual(Geometry.Ticker(0, 0, logarithmic=logarithmic).labels, ['0'])
-                    self.assertEqual(Geometry.Ticker(-1, -1, logarithmic=logarithmic).labels, ['-1'])
-                    self.assertEqual(Geometry.Ticker(-math.inf, math.inf, logarithmic=logarithmic).labels, ['0'])
-                    self.assertEqual(Geometry.Ticker(-math.nan, math.nan, logarithmic=logarithmic).labels, ['0'])
-                    self.assertEqual(Geometry.Ticker(-math.inf, 1, logarithmic=logarithmic).labels, ['0'])
-                    self.assertEqual(Geometry.Ticker(0, math.inf, logarithmic=logarithmic).labels, ['0'])
+            self.assertEqual(Geometry.Ticker(0, 0, logarithmic=logarithmic).labels, ['0'])
+            self.assertEqual(Geometry.Ticker(1, 1, logarithmic=logarithmic).labels, ['1'])
+            self.assertEqual(Geometry.Ticker(-1, -1, logarithmic=logarithmic).labels, ['-1'])
+            self.assertEqual(Geometry.Ticker(-math.inf, math.inf, logarithmic=logarithmic).labels, ['0'])
+            self.assertEqual(Geometry.Ticker(-math.nan, math.nan, logarithmic=logarithmic).labels, ['0'])
+            self.assertEqual(Geometry.Ticker(math.nan, 1, logarithmic=logarithmic).labels, ['0'])
+            self.assertEqual(Geometry.Ticker(-math.inf, 1, logarithmic=logarithmic).labels, ['0'])
+            self.assertEqual(Geometry.Ticker(0, math.inf, logarithmic=logarithmic).labels, ['0'])
 
     def test_ticker_produces_expected_labels(self):
         self.assertListEqual(Geometry.Ticker(0, 1e8, ticks=3, logarithmic=False).labels, ['0e+00', '5e+07', '1.0e+08'])
```

**Comment by Brow71189** [comment]
Not objections from my side. Looks good.

**Comment by cmeyer** [comment]
Merged 836dcf7c4d54fd9634dd3b4bcf549ad3417233c8
